RealmSwift/Query.swift

```diff
@@ -824,6 +824,14 @@ extension Query where T: _HasPersistedType, T.PersistedType: _QueryNumeric {
     }
 }
 
+public extension Query where T: OptionalProtocol, T.Wrapped: EmbeddedObject {
```

Comment:
Without a tag protocol or dedicated type this means that `geoWithin` can be used on any `EmbeddedObject`. We should expose something to allow conformity.

Reply:
@leemaguire we'll need to expose a protocol, or a type, and require the user to conform or inherit from it, which is something we don't want to at the moment given we probably gonna build our own type in the future. Also, developers can as well use IN with filter and NSPredicate which is not gonna need conformance and probably confuse the user. 
This feature is gonna be documented to show  (https://www.mongodb.com/docs/realm/sdk/dotnet/model-data/data-types/geospatials/#creating-a-geojson-compatible-class) and will throw an error on runtime for any value if it is done incorrectly, which probably the developer is going to catch before production. 
Adding extra docstrings to warn the user about this.